requirements.txt

```diff
@@ -1,13 +1,26 @@
-# Common Requirements
--r requirements_common.txt
-
-# Find older pytorch pip wheel from PyTorch Org website
---find-links https://download.pytorch.org/whl/torch_stable.html
-
-# Torch CPU Version Windows/Linux
-torch==1.8.0+cpu; sys_platform != "darwin"
-torchvision==0.9.0+cpu; sys_platform != "darwin"
-
-# Torch CPU Version macOS (macOS has no CPU/GPU version separation)
-torch==1.8.0; sys_platform == "darwin"
-torchvision==0.9.0; sys_platform == "darwin"
\ No newline at end of file
+# workarounds for pip 20.0.2's buggy dependency resolver
+# which doesn't account for cross-package constraints: https://pip.pypa.io/en/stable/user_guide/#changes-to-the-pip-dependency-resolver-in-20-3-2020
+# matplotlib pulls in the latest pyparsing, nibabel pulls in the latest packaging, and the two latests are in conflict.
+# > ERROR: packaging 21.2 has requirement pyparsing<3,>=2.0.2, but you'll have pyparsing 3.0.5 which is incompatible.
+# This forces the older pip, if the user has the older pip, to behave itself.
+# This is to specially support users on Ubuntu 20.04 LTS; when Ubuntu 22.04 LTS comes out, this can be removed.
+pyparsing<3,>=2.0.2
+csv-diff>=1.0
+loguru~=0.5
+joblib~=1.0
+matplotlib>=3.3.0
+nibabel~=3.2
+onnxruntime==1.7.0
+pandas~=1.1
+pillow>=7.0.0
+pybids>=0.14.0
+scikit-learn>=0.20.3
+scikit-image~=0.17
+seaborn~=0.11
+tensorboard>=1.15.0
+tqdm>=4.30
+scipy
+torchio>=0.18.68
+torch~=1.8.1
+torchvision~=0.9.1
```

Comment:
Just to make sure I'm understanding correctly, I've written a quick first draft of what a theoretical docs update would look like:

> # Installing ivadomed
> 
> ## Default installation
> 
> You can quickly get up and running by using the following command to install `ivadomed`:
> 
> ```
> pip install ivadomed
> ```
> 
> Out of the box, this installation command will use the following defaults:
> 
> * Linux users: GPU processing (CUDA 10.2)
> * Windows users: CPU-only processing
> * macOS users: CPU-only processing
> 
> ## Installation special cases
> 
> If these defaults don't suit your needs, please instead use choose one of the installation methods below:
> 
> ### GPU processing (macOS, Windows, Linux)
> 
> First, you will need to determine the version of CUDA installed on your system by using the following command:
> 
> ```
> nvidia-smi
> ```
> 
> Then, you can fill in the CUDA version by using the following command to install ivadomed:
> 
> ```
> pip install ivadomed --extra-index-url https://download.pytorch.org/whl/cuXXX
> ```
> 
> For example, if the `nvidia-smi` command says "`CUDA Version: 11.2`", you will want to use the following command:
> 
> ```
> pip install ivadomed --extra-index-url https://download.pytorch.org/whl/cu112
> ```
> 
> ### CPU-only processing (Linux)
> 
> If you are using Linux, and would like to reduce the size of your installation by limiting processing to the CPU only, you can use the following command to install `ivadomed`:
> 
> ```
> pip install ivadomed --extra-index-url https://download.pytorch.org/whl/cpu
> ```

Is this information accurate? Are these the right defaults, and the right installation instructions?

Reply:
And for https://download.pytorch.org/whl/cpu, we probably don't even need to explain that it's Linux-specific. It is at the moment but maybe that'll change; I was thinking we could just say "if you know you don't have a GPU, use --extra-index-url https://download.pytorch.org/whl/cpu to save some space".